modules/core/src/main/java/org/apache/ignite/IgniteSnapshot.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite;
+
+import java.util.List;
+import org.apache.ignite.lang.IgniteFuture;
+
+/**
+ * This interface provides functionality for creating cluster-wide cache data snapshots.
+ * <p>
+ * Current limitations:
+ * <ul>
+ * <li>Snapshot will trigger PME (partition map exchange) to run itself.</li>
+ * <li>Snapshot will be taken from all registered persistence caches to
+ * grantee data consistency between them.</li>
+ * <li>Snapshot must be resorted manually on the switched off cluster by copying data
+ * to the working directory on each cluster node.</li>
+ * </ul>
+ */
+public interface IgniteSnapshot {
+    /**
+     * @return List of all known snapshots.
+     */
+    public List<String> getSnapshots();
```

Comment:
Should we return a object instead of a String if we add more information about snapshot in future?

Reply:
For the initial create snapshot procedure I'd like to focus only on locally available information about snapshots on each node. So, I've renamed this method to `getSnapshotNamesLocal` which will return all snapshot names on the local node. 

If we decide to work further and additional snapshot meta needs to be collected prior to a snapshot creation. The API can be extended with, for instance: 
- `public List<SnapshotMeta> getSnapshots();` -- globally known snapshots.
- `snapshots` view with global snapshots

Reply:
I've decided to remove from java API this method since it has no sense. I think it's better to extend API when the snapshots view and cluster-wide information about snapshot will be implemented.

`LocalSnapshotNames` are available through the JMX.